README.md

```diff
@@ -1,2 +1,53 @@
-# example-application
-Example out-of-tree application that is also a module
+# Zephyr Example Application
+
+This repository contains a Zephyr example application. The main purpose of this
+repository is to serve as a reference on how to structure Zephyr based
+applications. Some of the features demonstrated in this example are:
+
+- Basic application skeleton
+- Custom boards
+- Custom devicetree bindings
+- Out-of-tree drivers
+- Documentation using Doxygen and Sphinx
```

Comment:
Maybe some links to help the reader to the relevant parts in Zephyr docs, like drivers, board porting.
https://docs.zephyrproject.org/latest/reference/drivers/index.html
https://docs.zephyrproject.org/latest/guides/porting/board_porting.html

Reply:
done